WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderViewModel.kt

```diff
@@ -116,9 +118,24 @@ class ReaderViewModel @Inject constructor(
     fun onTagChanged(selectedTag: ReaderTag?) {
         // Store most recently selected tab so we can restore the selection after restart
         appPrefsWrapper.setReaderTag(selectedTag)
+
+        // Show interests picker if tag changed to discover and no followed tags found for user
+        if (selectedTag?.isDiscover == true &&
+                appPrefsWrapper.isReaderImprovementsPhase2Enabled()) {
+            launch {
+                val userTags = getFollowedTagsUseCase.get()
+                if (userTags.isEmpty()) {
+                    isReaderInterestsShown = true
+                    initialized = false
+                    _uiState.value = InitialUiState
+                    _showReaderInterests.value = Event(Unit)
+                }
```

Comment:
This comment is not related to this PR: Flags like "initialized" (with the current usage) and "isReaderInterestsShown" might not be maintainable in the long run. Imagine we keep adding such flags - the class would become complex spaghetti code (especially since some combinations of these flags might not be valid -> eg "initialized = false, isReaderInterestsShown = true").
I'm wondering if we could simply turn these flags into UiStates. Technically this screen always goes through these states:  [Display interests screen] -> loadTabs -> Initialize tabs -> show content. 

Any thoughts? If you think it's worth considering, could you please create a note on the project board and move it to the last position in the "Android Backlog" column. Thanks!

Reply:
Good advice, thanks! Note added [here](https://github.com/orgs/wordpress-mobile/projects/63#card-43190584).